verifier/src/advisories.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+extern crate alloc;
+
+use crate::{Accessor, SpacedStructName, VerificationOutput, Verifier};
+use alloc::collections::BTreeSet;
+use alloc::string::{String, ToString};
+use core::fmt::{Display, Formatter};
+use serde::Deserialize;
+
+/// The status of a set of advisories
+///
+/// The variants are defined in the schema at
+/// <https://api.portal.trustedservices.intel.com/documentation#pcs-tcb-info-model-v3>
+///
+/// The variant order is important here, the higher the index the better the
+/// status. For example: `UpToDate` is a better status than `SWHardeningNeeded`.
+/// ```
+/// use mc_attestation_verifier::AdvisoryStatus;
+/// assert!(AdvisoryStatus::UpToDate > AdvisoryStatus::SWHardeningNeeded);
+/// ```
+#[derive(Debug, PartialEq, Eq, PartialOrd, Hash, Copy, Clone, Deserialize)]
+pub enum AdvisoryStatus {
```

Comment:
IFF this naming makes sense we may want to revist the TCB names, I was more or less copying the JSON field names.

I think to the client the status for the "advisory" is cleaner in that's what the client will be setting up the "TCB" values will mostly be behind the scenes for them.

I originally had this as `Status` but with `XVerifier` everywhere it seemed fine to namespace this struct as well